packages/flutter_tools/lib/src/project.dart

```diff
@@ -881,6 +881,41 @@ $javaGradleCompatUrl
     }
     return AndroidEmbeddingVersionResult(AndroidEmbeddingVersion.v1, 'No `<meta-data android:name="flutterEmbedding" android:value="2"/>` in ${appManifestFile.absolute.path}');
   }
+
+  // TODO(matanlurey): Flip to true when on by default, https://github.com/flutter/flutter/issues/132712.
+  static const bool _impellerEnabledByDefault = false;
+
+  /// Returns the `io.flutter.embedding.android.EnableImpeller` manifest value.
+  ///
+  /// If there is no manifest file, or the key is not present, returns `false`.
+  bool computeImpellerEnabled() {
+    if (!appManifestFile.existsSync()) {
+      return _impellerEnabledByDefault;
+    }
+    final XmlDocument document;
+    try {
+      document = XmlDocument.parse(appManifestFile.readAsStringSync());
+    } on XmlException {
+      throwToolExit('Error parsing $appManifestFile '
+                    'Please ensure that the android manifest is a valid XML document and try again.');
+    } on FileSystemException {
+      throwToolExit('Error reading $appManifestFile even though it exists. '
+                    'Please ensure that you have read permission to this file and try again.');
+    }
```

Comment:
uber-nit: Handling `FileSystemException` is _probably_ redundant as `appManifestFile` should be an instance of `ErrorHandlingFile`[^1], which should [handle permissions issues](https://github.com/flutter/flutter/blob/4a65a76279cda479d0c8a2a57ac8f34890e1adf2/packages/flutter_tools/lib/src/base/error_handling_io.dart#L595-L599), but this is fine.

[^1]: this is the case of all instances of `File` within the flutter tool code, though we occasionally get surprised by exceptions coming from instances of `File` rather than `ErrorHandlingFile`

Reply:
Just a fun fact: we've had instances where `existsSync` returns `true`, but an immediately-following synchronous file op fails with a file/path-not-found error 🙃. As a result—for some tiny amount of users—this exit message could be inaccurate 😛